From 9d5f0a95b766536830a6bab961c77a11bde7f477 Mon Sep 17 00:00:00 2001
From: William Jordan <bjordan@habana.ai>
Date: Fri, 16 May 2025 21:51:35 +0300
Subject: [PATCH] [SW-218976] add entity aux names to runtime

Add Entity AUX Name PRRs to the arcmgmt PLDM
PDR database.

Change-Id: I8bd941a3e9f124f9dfe6bf2070904b713a530056
Signed-off-by: William Jordan <bjordan@habana.ai>
---

diff --git a/arch/gaudi3/ras/pldm/pldm_pdr_generated.c b/arch/gaudi3/ras/pldm/pldm_pdr_generated.c
index 6a0b844..d108df2 100644
--- a/arch/gaudi3/ras/pldm/pldm_pdr_generated.c
+++ b/arch/gaudi3/ras/pldm/pldm_pdr_generated.c
@@ -1,16 +1,16 @@
 // SPDX-License-Identifier: Apache-2.0
 /*
- * Copyright (c) 2024 Habana Labs
+ * Copyright (c) 2025 Habana Labs
  */
 
 /* !!! AUTO-GENERATED FILE - DO NOT EDIT !!! */
 /*
- * Spreadsheet : '20250507_GAUDI3_RAS_PLDM_7th_stage.xlsx'
- * MD5         : 6bbcc11a6d64ea92876ded4f307e00b4
+ * Spreadsheet : '20250504_GAUDI3_RAS_PLDM_7th_stage.xlsx'
+ * MD5         : f4a4d7a684a412cfb628b033771747ea
  *
  * Command line:
  *   ./g3-gen-pdrs.py \
- *     -w ".../20250507_GAUDI3_RAS_PLDM_7th_stage.xlsx" \
+ *     -w ".../20250504_GAUDI3_RAS_PLDM_7th_stage.xlsx" \
  *     -r ".../pldm_pdr.h" "pdrs" \
  *     --header ".../pldm_pdr_generated.h" \
  *     --source ".../pldm_pdr_generated.c"
@@ -4095,6 +4095,151 @@
 };
 const size_t pldm_pdr_name_records_length = ARRAY_SIZE(pldm_pdr_name_records);
 
+/**** Auxilary Name PDRs ****/
+
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+
+/* Gaudi3 OAM add-in card accelerator */
+#define _oamAddInCardAccelerator_str "oamAddInCardAccelerator"
+/* Gaudi3 OAM CPLD */
+#define _oamCPLD_str "oamCPLD"
+/* Gaudi3 accelerator */
+#define _accelerator_str "accelerator"
+/* Gaudi3 ethernet port */
+#define _ethernetPort_str "ethernetPort"
+
+#else /* old aux name format */
+
+/* Gaudi3 OAM add-in card accelerator */
+const uint16_t _oamAddInCardAccelerator_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6f00, 0x6100, 0x6d00, 0x4100, 0x6400, 0x6400,
+	0x4900, 0x6e00, 0x4300, 0x6100, 0x7200, 0x6400,
+	0x4100, 0x6300, 0x6300, 0x6500, 0x6c00, 0x6500,
+	0x7200, 0x6100, 0x7400, 0x6f00, 0x7200, 0x0000
+};
+/* Gaudi3 OAM CPLD */
+const uint16_t _oamCPLD_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6f00, 0x6100, 0x6d00, 0x4300, 0x5000, 0x4c00,
+	0x4400, 0x0000
+};
+/* Gaudi3 accelerator */
+const uint16_t _accelerator_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6100, 0x6300, 0x6300, 0x6500, 0x6c00, 0x6500,
+	0x7200, 0x6100, 0x7400, 0x6f00, 0x7200, 0x0000
+};
+/* Gaudi3 ethernet port */
+const uint16_t _ethernetPort_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6500, 0x7400, 0x6800, 0x6500, 0x7200, 0x6e00,
+	0x6500, 0x7400, 0x5000, 0x6f00, 0x7200, 0x7400,
+	0x0000
+};
+
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+
+struct pldm_pdr_entity_name_record pldm_pdr_entity_name_records[] OFFLOAD_PDR_ATTR = {
+	[PLDM_RIDX_ENTITY_AUX_NAMES_ADDIN_CARD] = {
+		.entity_aux_names = {
+			.header = {
+				.record_handle = PLDM_RH_ENTITY_AUX_NAMES_ADDIN_CARD,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_ENTITY_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.entity_type = PLDM_ENTITY_TYPE_ADDIN_CARD,
+			.entity_instance = 1,
+			.entity_container_id = PLDM_CONTAINER_ADDIN_CARD,
+			.shared_name_count = 0,
+			.names = {
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+				_oamAddInCardAccelerator_str,
+#else /* old aux name format */
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _oamAddInCardAccelerator_str,
+				},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+			},
+		},
+	},
+	[PLDM_RIDX_ENTITY_AUX_NAMES_OAM_CPLD] = {
+		.entity_aux_names = {
+			.header = {
+				.record_handle = PLDM_RH_ENTITY_AUX_NAMES_OAM_CPLD,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_ENTITY_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.entity_type = PLDM_ENTITY_TYPE_CPLD,
+			.entity_instance = 1,
+			.entity_container_id = PLDM_CONTAINER_ADDIN_CARD,
+			.shared_name_count = 0,
+			.names = {
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+				_oamCPLD_str,
+#else /* old aux name format */
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _oamCPLD_str,
+				},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+			},
+		},
+	},
+	[PLDM_RIDX_ENTITY_AUX_NAMES_ACCELERATOR] = {
+		.entity_aux_names = {
+			.header = {
+				.record_handle = PLDM_RH_ENTITY_AUX_NAMES_ACCELERATOR,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_ENTITY_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.entity_type = PLDM_ENTITY_TYPE_ACCELERATOR,
+			.entity_instance = 1,
+			.entity_container_id = PLDM_CONTAINER_ADDIN_CARD,
+			.shared_name_count = 0,
+			.names = {
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+				_accelerator_str,
+#else /* old aux name format */
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _accelerator_str,
+				},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+			},
+		},
+	},
+	[PLDM_RIDX_ENTITY_AUX_NAMES_ETHERNET_PORT] = {
+		.entity_aux_names = {
+			.header = {
+				.record_handle = PLDM_RH_ENTITY_AUX_NAMES_ETHERNET_PORT,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_ENTITY_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.entity_type = PLDM_ENTITY_TYPE_ETHERNET,
+			.entity_instance = 1,
+			.entity_container_id = PLDM_CONTAINER_ACCELERATOR,
+			.shared_name_count = 23,
+			.names = {
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+				_ethernetPort_str,
+#else /* old aux name format */
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _ethernetPort_str,
+				},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+			},
+		},
+	},
+};
+const size_t pldm_pdr_entity_name_records_length = ARRAY_SIZE(pldm_pdr_entity_name_records);
+
 /**** OEM State Set PDRs ****/
 
 #if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
diff --git a/include/arch/gaudi3/pldm_pdr.h b/include/arch/gaudi3/pldm_pdr.h
index cadd0ec..a4fbbeb 100644
--- a/include/arch/gaudi3/pldm_pdr.h
+++ b/include/arch/gaudi3/pldm_pdr.h
@@ -378,6 +378,10 @@
 	PLDM_RH_EFFECTER_ERROR_INJECTION_HBMS				= 3088,
 	PLDM_RH_EFFECTER_ERROR_INJECTION_TDR				= 3089,
 	PLDM_RH_EFFECTER_ERROR_INJECTION_BOOT				= 3090,
+	PLDM_RH_ENTITY_AUX_NAMES_ADDIN_CARD				= 4000,
+	PLDM_RH_ENTITY_AUX_NAMES_OAM_CPLD				= 4001,
+	PLDM_RH_ENTITY_AUX_NAMES_ACCELERATOR				= 4002,
+	PLDM_RH_ENTITY_AUX_NAMES_ETHERNET_PORT				= 4003,
 	PLDM_RH_SENSOR_AUX_NAMES_CPLD_FW_VERSION			= 4010,
 	PLDM_RH_SENSOR_AUX_NAMES_ASIC_SERIAL_NUM_HIGH			= 4011,
 	PLDM_RH_SENSOR_AUX_NAMES_ASIC_SERIAL_NUM_LOW			= 4012,
diff --git a/include/arch/gaudi3/pldm_pdr_generated.h b/include/arch/gaudi3/pldm_pdr_generated.h
index 60fe52c..04979fe 100644
--- a/include/arch/gaudi3/pldm_pdr_generated.h
+++ b/include/arch/gaudi3/pldm_pdr_generated.h
@@ -1,16 +1,16 @@
 /* SPDX-License-Identifier: Apache-2.0
  *
- * Copyright (c) 2024 Habana Labs
+ * Copyright (c) 2025 Habana Labs
  */
 
 /* !!! AUTO-GENERATED FILE - DO NOT EDIT !!! */
 /*
- * Spreadsheet : '20250507_GAUDI3_RAS_PLDM_7th_stage.xlsx'
- * MD5         : 6bbcc11a6d64ea92876ded4f307e00b4
+ * Spreadsheet : 'GAUDI3 RAS PLDM_7th_stage.xlsx'
+ * MD5         : 28e55620fbf641fdc77b9a648ffea672
  *
  * Command line:
- *   ./g3-gen-pdrs.py \
- *     -w ".../20250507_GAUDI3_RAS_PLDM_7th_stage.xlsx" \
+ *   /home/bjordan/trees/npu-stack/embedded/zephyr-modules/pldm/scripts/g3-gen-pdrs.py \
+ *     -w ".../GAUDI3 RAS PLDM_7th_stage.xlsx" \
  *     -r ".../pldm_pdr.h" "pdrs" \
  *     --header ".../pldm_pdr_generated.h" \
  *     --source ".../pldm_pdr_generated.c"
@@ -107,6 +107,13 @@
 	PLDM_RIDX_EFFECTER_AUX_NAMES_LIMIT_TDP_12V,
 };
 
+enum pldm_entity_aux_ridx {
+	PLDM_RIDX_ENTITY_AUX_NAMES_ADDIN_CARD,
+	PLDM_RIDX_ENTITY_AUX_NAMES_OAM_CPLD,
+	PLDM_RIDX_ENTITY_AUX_NAMES_ACCELERATOR,
+	PLDM_RIDX_ENTITY_AUX_NAMES_ETHERNET_PORT,
+};
+
 enum pldm_oem_state_set_ridx {
 	PLDM_RIDX_SENSOR_OEM_STATE_SET_ETHERNET_MAP,
 	PLDM_RIDX_SENSOR_OEM_STATE_SET_FW_STATE,
@@ -140,6 +147,9 @@
 extern struct pldm_pdr_name_record pldm_pdr_name_records[];
 extern const size_t pldm_pdr_name_records_length;
 
+extern struct pldm_pdr_entity_name_record pldm_pdr_entity_name_records[];
+extern const size_t pldm_pdr_entity_name_records_length;
+
 extern struct pldm_pdr_oem_state_set_record pldm_pdr_oem_state_set_records[];
 extern const size_t pldm_pdr_oem_state_set_records_length;
 
diff --git a/lib/ras/pldm/pldm.c b/lib/ras/pldm/pldm.c
index 608aa0c..2d8c06d 100644
--- a/lib/ras/pldm/pldm.c
+++ b/lib/ras/pldm/pldm.c
@@ -75,6 +75,8 @@
 		.num_records = pldm_pdr_records_length,
 		.name_records = pldm_pdr_name_records,
 		.num_name_records = pldm_pdr_name_records_length,
+		.entity_name_records = pldm_pdr_entity_name_records,
+		.num_entity_name_records = pldm_pdr_entity_name_records_length,
 		.oem_state_set_records = pldm_pdr_oem_state_set_records,
 		.num_oem_state_set_records = pldm_pdr_oem_state_set_records_length,
 		.oem_unit_records = pldm_pdr_oem_unit_records,
